### PR TITLE
Add initial version of C++ documentation

### PR DESCRIPTION
**Pull request by doosuu**
## Describe your changes

Extended documentation with relevant parts for the C++ SDK.

## Issue ticket number and link
https://softwaredefinedvehicle.visualstudio.com/SoftwareDefinedVehicle/_workitems/edit/10702

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] Do we need to implement analytics?
- [x] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

File: content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/_index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        python3 gen_vehicle_model.py -I ./vehicle_signal_specification/spec ./vehicle_signal_specification/spec/VehicleSignalSpecification.vspec -l <lang> -T sdv_model -N sdv_model
+        ```
+
+        Depending on the value of `lang`, which can assume the values `python` and `cpp`, this creates a `sdv_model` directory in the root of repository along with all generated source files for the given programming language.
+
+        Here is an overview of what is generated for every available value of `lang`:
+        | lang   | output                                                                        |
+        | -------|------------------------------------------------------------------------------ |
+        | python | python sources and a `setup.py` ready to be used as python package            |
+        | cpp    | c++ sources, headers and a CMakeLists.txt ready to be used as a CMake project |
```
Comment (MP91) [inline content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/_index.md:58]: the output is not very readable, please add some space between the columns

File: content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/_index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        To have a custom model name, refer to README of [vehicle-model-generator](https://github.com/eclipse-velocitas/vehicle-model-generator) repository.
+  4. For python: Change the version of package in `setup.py` manually (defaults to 0.1.0).
+  5. Now the newly generated `sdv_model` can be used for distribution. (See [Distributing your Python Vehicle Model](#distributing-your-python-vehicle-model))
```
Comment (MP91) [inline content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/_index.md:62]: please adjust the link

File: content/en/docs/tutorials/tutorial_how_to_deploy_a_vehicle_app_with_helm.md [hunk 3]
```diff
@@ -12,16 +12,16 @@ aliases:
...
 
 - Prepare a Helm chart
-- Deploy your Python Vehicle App to local K3D
+- Deploy your Vehicle App to local K3D
 
 ## Prerequisites
 
 - [Visual Studio Code](https://code.visualstudio.com/) with the [Python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python) installed. For information on how to install extensions on Visual Studio Code, see [VS Code Extension Marketplace](https://code.visualstudio.com/docs/editor/extension-gallery).
-- Completed the tutorial [How to create a vehicle app](/docs/python-sdk/tutorial_how_to_create_a_vehicle_app.md)
+- Completed the tutorial [How to create a vehicle app]({{< ref "/docs/tutorials/vehicle-app-development" >}})
```
Comment (MP91) [inline content/en/docs/tutorials/tutorial_how_to_deploy_a_vehicle_app_with_helm.md:20]: please use plain md if possible

File: content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_python.md [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
-  Learn how to develop and test the Vehicle App.
+  Learn how to develop and test the Vehicle App using Python.
 aliases:
-  - /docs/tutorials/tutorial_how_to_create_a_vehicle_app.md
-  - /docs/python-sdk/tutorial_how_to_create_a_vehicle_app.md
+  - /docs/tutorials/python/tutorial_how_to_create_a_vehicle_app_python.md
 ---
 
-> We recommend that you make yourself familiar with the [Python Vehicle App SDK](/python-sdk/python_vehicle_app_sdk_overview.md) first, before going through this tutorial.
+> We recommend that you make yourself familiar with the [Vehicle App SDK]({{< ref "vehicle_app_sdk" >}} "SDK Overview") first, before going through this tutorial.
```
Comment (MP91) [inline content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_python.md:11]: please use plain md

**Review by MP91 — CHANGES_REQUESTED** [review]

File: content/en/docs/concepts/vehicle_app_sdk/_index.md [hunk 5]
```diff
@@ -130,20 +127,92 @@ class Vehicle(Model):
...
 
 ### Model integration
 
-Based on the `.proto` files of the Vehicle Services, the protocol buffers compiler generates special Python descriptors for all rpcs, messages, fields etc.
+Based on the `.proto` files of the Vehicle Services, the protocol buffers compiler generates descriptors for all rpcs, messages, fields etc for the target language.
 The gRPC stubs are wrapped by a **convenience layer** class derived from `Service` that contains all the methods of the underlying protocol buffer specification.
 
-```Python
+{{% alert title="Info" %}}
+The convencience layer of C++ is abit more extensive than in Python. The complexity of gRPC's async API is hidden behind individual `AsyncGrpcFacade` implementations which need to be implemented manually. Have a look at the `SeatAdjusterApp` example's `SeatService` and its `SeatSeatServiceAsyncGrpcFacade`.
```
Comment (dennismeister93) [inline content/en/docs/concepts/vehicle_app_sdk/_index.md:211]:
```suggestion
The convencience layer of C++ is abit more extensive than in Python. The complexity of gRPC's async API is hidden behind individual `AsyncGrpcFacade` implementations which need to be implemented manually. Have a look at the `SeatAdjusterApp` example's `SeatService` and its `SeatServiceAsyncGrpcFacade`.
```

File: content/en/docs/concepts/vehicle_app_sdk/_index.md [hunk 6]
```diff
@@ -235,46 +380,70 @@ The SDK supports publishing messages to a MQTT broker and subscribing to topics
...
-```
+{{< /tab >}}
+{{< tab "C++" >}}
+publishToTopic("seatadjuster/currentPosition", "{ \"position\": 40 }");
+{{< /tab >}}
+{{< /tabpane >}}
 
 ### Subscribe to MQTT Topics
 
 Subscriptions to MQTT topics can be easily established with the `subscribe_topic()` annotation. The annotation needs to be applied to a method of the `Vehicle App` class.
```
Comment (dennismeister93) [inline content/en/docs/concepts/vehicle_app_sdk/_index.md:395]: i think the annotation implementation is only valid for python

File: content/en/docs/concepts/vehicle_app_sdk/_index.md [hunk 7]
```diff
@@ -291,7 +460,30 @@ LOOP = asyncio.get_event_loop()
...
+{{< tab "C++" >}}
+#include "VehicleApp.h"
+#include "vehicle_model/Vehicle.h"
+
+using namespace velocitas;
+
+class SeatAdjusterApp : public VehicleApp {
+public:
+    SeatAdjusterApp()
+        : VehicleApp(IVehicleAdataBrokerClient::createInstance("vehicledatabroker")),
```
Comment (dennismeister93) [inline content/en/docs/concepts/vehicle_app_sdk/_index.md:473]: Is the A a typo in "IVehicleAdataBrokerClient" ?

File: content/en/docs/tutorials/quickstart.md [hunk 8]
```diff
@@ -9,15 +9,22 @@ aliases:
...
+{{< tabpane text=true >}}
+{{% tab header="Python" %}}
+Create your own repository copy from the [Template Repository](https://github.com/eclipse-velocitas/vehicle-app-python-template) by clicking the green button `Use this template`. You don't have to include all branches. For more information on Template Repositories take a look at this [GitHub Tutorial](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template#creating-a-repository-from-a-template).
+{{% /tab %}}
+{{% tab header="C++" %}}
+C++ currently has only an [SDK repository](https://github.com/eclipse-velocitas/vehicle-app-cpp-sdk). To create your own _Vehicle App_ repository, fork the repository into your own organization. For the fork, only the `main` branch needs to be copied over.
+{{% /tab %}}
+{{< /tabpane >}}
 
 ## Starting Development Environement
```
Comment (dennismeister93) [inline content/en/docs/tutorials/quickstart.md:29]:
```suggestion
## Starting Development Environment
```

File: content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/_index.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
+---
+title: "Vehicle Model Creation"
+date: 2022-05-09T13:43:25+05:30
+weight: 1
+description: >
+  Learn how to create a Vehicle Model to get to access vehicle data or execute remote procedure calls.
```
Comment (dennismeister93) [inline content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/_index.md:6]:
```suggestion
  Learn how to create a Vehicle Model to get access to vehicle data or execute remote procedure calls.
```

File: content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/distribution_python.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  Learn how to distribute a Vehicle Model written in Python.
+aliases:
+  - /docs/tutorials/how_to_create_a_vehicle_model/distribution_python.md
+---
+
+Now you a have a Python package containing your first Python _Vehicle Model_ and it is time to distribute it. There is nothing special about the distribution of this package, since it is just an ordinary Python package. Check out the [Python Packaging User Guide](https://packaging.python.org/en/latest/) to learn more about packaging and package distribution in Python.
+
+## Distribute to single Vehicle App
+
+If you want to distribute your Python _Vehicle Model_ to a single _Vehicle App_, you can do so by copying the entire folder `my_vehicle_model` under the `src` folder of your _Vehicle App_ repository and treat is a sub-package of the _Vehicle App_.
```
Comment (dennismeister93) [inline content/en/docs/tutorials/tutorial_how_to_create_a_vehicle_model/distribution_python.md:15]:
```suggestion
If you want to distribute your Python _Vehicle Model_ to a single _Vehicle App_, you can do so by copying the entire folder `my_vehicle_model` under the `src` folder of your _Vehicle App_ repository and treat it as a sub-package of the _Vehicle App_.
```

File: content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+weight: 2
+description: >
+  Learn how to develop and test a Vehicle App using C++.
+aliases:
+  - /docs/tutorials/python/tutorial_how_to_create_a_vehicle_app_cpp.md
+---
+
+> We recommend that you make yourself familiar with the [Vehicle App SDK]({{< ref "vehicle_app_sdk" >}} "SDK Overview") first, before going through this tutorial.
+
+The following information describes how to develop and test the sample Vehicle App that is included in the [template repository](https://github.com/eclipse-velocitas/vehicle-app-python-template). You will learn how to use the Vehicle App SDK and how to interact with the Vehicle Model.
```
Comment (dennismeister93) [inline content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md:13]: Link refers to python template

File: content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+
+Once you have completed all steps, you will have a solid understanding of the development workflow and you will be able to reuse the template repository for your own _Vehicle App_ development project.
+
+## Develop your first Vehicle App
+
+This section describes how to develop your first _Vehicle App_. Before you start building a new _Vehicle App_, make sure you have already read the other manuals:
+
+- [Setup and Explore Development Enviroment](/docs/setup_and_explore_development_environment.md)
+- [How to create a Vehicle Model](/docs/python-sdk/tutorial_how_to_create_a_vehicle_model.md)
```
Comment (dennismeister93) [inline content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md:22]: This link ends up in 404

File: content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md [hunk 13]
```diff
@@ -0,0 +1,367 @@
...
+
+The app is now running. In order to use it properly, we will enhance the app with more features in the next sections.
+
+## Vehicle Model
+
+In order to facilitate the implementation, the whole vehicle is abstracted into model classes. Please check [tutorial about creating models]({{< ref "docs/tutorials/tutorial_how_to_create_a_vehicle_model" >}}) for more details about this topic. In this section, the focus is on using the models.
+
+### Import the model
+
+The first thing you need to do to get access to the Vehicle Model. In the section about [distributing a model]({{< ref "tutorial_how_to_create_a_vehicle_model.md#distributing-your-python-vehicle-model" >}}), you got to know the different methods.
```
Comment (dennismeister93) [inline content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md:99]:
```suggestion
The first thing you need to do to get access to the Vehicle Model. In the section about [distributing a model]({{< ref "tutorial_how_to_create_a_vehicle_model.md#distributing-your-vehicle-model" >}}), you got to know the different methods.
```

File: content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
+                "ignoreFailures": true
+            }
+        ],
+        "preLaunchTask": "dapr-VehicleApp-run",
+        "postDebugTask": "dapr-VehicleApp-stop"
+    }
+]
+```
+
+We specify which python-script to run using the `program` key. With the `preLaunchTask` and `postDebugTask` keys, you can also specify tasks to run before or after debugging. In this example, DAPR is set up to start the app before and stop it again after debugging. Below you can see the 2 tasks.
```
Comment (dennismeister93) [inline content/en/docs/tutorials/vehicle-app-development/tutorial_how_to_create_a_vehicle_app_cpp.md:321]: it's not a python script here

**Review by MP91 — APPROVED** [review]

**Pull request merged**